aws/lambda/lambda.tf

```diff
@@ -10,8 +10,8 @@ data "external" "builder" {
 data "archive_file" "function_zip" {
   type        = "zip"
   output_path = coalesce(var.archive.output_path, "dist/${var.name}/lambda.zip")
-  source_dir  = var.source_dir
-  excludes    = var.is_bundle ? concat(var.archive.excludes, ["**/node_modules/**", "**/.yarn/**"]) : var.archive.excludes
+  source_dir  = var.is_bundle ? "${var.source_dir}/dist" : var.source_dir
```

Comment:
Is this a breaking change for dependants of they-ray? Looks like we are changing the file path when moving `source_dir`?

Reply:
did you mean they-terraform instead of they-ray?

Reply:
> did you mean they-terraform instead of they-ray?

Yes 🫠

Reply:
All this does is:
- if we are bundling (which we expect to result in a `dist` dir), we only copy the files inside `dist` dir. Which is only the index.js, which is all that is needed for our lambda zip deployment.

We could think about somehow specifing the `bundle_output_dir` but I feel like `dist` is a standard convention.

Also, they-ray does not use they-terraform modules (yet?) so this should not be an issue.

Reply:
I believe @e2b meant to say, is this a breaking change for dependants of they-terraform :P 
Regardless your explanation for why this isn't a problem seems sufficient to that question from my perspective @xddq.
Basically if any dependent of they-terraform that is using bundling, up to now was using an "overloaded" lambda zip, and this change just means that gets rectified.

Reply:
Fine for me. Looks like we are already using `dist/` folders, just don't know how we referred to that folder before.

https://github.com/THEY-Consulting/cp-api/blob/079f46b4b94a37de51fe8f21e445a72a522d4480/terraform/modules/endpoints/get-tariff-details/lambda.tf#L2-L12

Reply:
Because function apps need more than the dist folder this wouldn't work for the azure module. This would mean `lambda.is_bundle` would behave differently than `function_app.is_bundle`?

Reply:
But the change in `aws/lambda/lambda.tf` only affects the aws part, would it be so problematic, if aws and azure behaved differently in what gets used from the dist folder, depending on what they need? Might not see the problem, but I don't see how they need to behave the same way, when they need different things.
Another note: I am wondering, if the globs/wildcard support doesn't work, than what we have also doesn't properly work for azure either no?

Reply:
you are right, it probably doesn't matter if the internal behaviour is different.

Reply:
Maybe we need to move the archive logic into the build.sh script as well? Hopefully there we can implement globs/wildcard correctly? 🤔

Reply:
also are we sure, we never need the package.json or anything outside the dist folder?

Reply:
> Because function apps need more than the dist folder this wouldn't work for the azure module. This would mean lambda.is_bundle would behave differently than function_app.is_bundle?

@thiemogries yes, azure module will need another internal implementation to fix this.

> Fine for me. Looks like we are already using dist/ folders, just don't know how we referred to that folder before.

@e2b We did not, we just put everything inside src into the zip and tried to exlude node_modules, which did not work. I also wondered if we even used the bundled file in the lambda before. Not sure to be fair.

@e2b Please resolve if you think this is done, I think this can be resolved.